ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -652,9 +658,21 @@ let on_request :
     let+ doc = Document_store.get_opt store uri in
     Document.syntax doc
   in
-  match syntax with
-  | Some (Ocamllex | Menhir) -> not_supported ()
-  | _ -> ocaml_on_request server req
+  match req with
+  | Client_request.UnknownRequest { meth; params } -> (
+    match
+      [ (Switch_impl_intf.meth, Switch_impl_intf.on_request) ]
+      |> List.assoc_opt meth
+    with
```

Comment:
I wonder if using extensible GADTs to represent requests and responses would result in more elegant handling of lsp implementation-specific requests.

Reply:
Could be. But keep in mind that requests are server specific, so it's not completely correct to extend the request type in general. This would cause problems for those developing more than one server at a time or combining servers somehow.